File: Emulator/src/main/content/region/kandarin/quest/fishingcompo/FishingContest.java
package content.region.kandarin.quest.fishingcompo;

import core.game.node.entity.player.Player;
import core.game.node.entity.player.link.quest.Quest;
import core.game.node.entity.skill.Skills;
import core.game.node.item.Item;
import core.plugin.Initializable;

import static core.api.ContentAPIKt.removeAttribute;

/**
 * The type Fishing contest.
 */
@Initializable
public class FishingContest extends Quest {

    /**
     * Instantiates a new Fishing contest.
     */
    public FishingContest() {
        super("Fishing Contest", 62, 61, 1, 11, 0, 1, 5);
    }

    /**
     * The constant FISHING_ROD.
     */
    public static final Item FISHING_ROD = new Item(307);

    /**
     * The constant FISHING_PASS.
     */
    public static final Item FISHING_PASS = new Item(27);

    /**
     * The constant RED_VINE_WORM.
     */
    public static final Item RED_VINE_WORM = new Item(25);

    /**
     * The constant RAW_GIANT_CARP.
     */
    public static final Item RAW_GIANT_CARP = new Item(338);

    /**
     * The constant GIANT_CARP.
     */
    public static final Item GIANT_CARP = new Item(337);

    /**
     * The constant FISHING_TROPHY.
     */
    public static final Item FISHING_TROPHY = new Item(26);

    /**
     * The constant GARLIC.
     */
    public static final Item GARLIC = new Item(1550);

    /**
     * The constant SPADE.
     */
    public static final Item SPADE = new Item(952);

    @Override
    public void drawJournal(Player player, int stage) {
        int line = 11;
        super.drawJournal(player, stage);
        if (stage < 10) {
            line(player, "I can start this quest by trying to take the !!shortcut??", line++);
            line(player, "near !!White Wolf Mountain??", line++);
            line(player, "I need level !!10 Fishing?? to start this quest.", line++, player.getSkills().getLevel(Skills.FISHING) >= 10);
        } else if (stage >= 10) {
            line(player, "The !!mountain Dwarves' home?? would be an ideal way to get across ", line++, stage >= 20);
            line(player, "White Wolf Mountain safely. However, the Dwarves aren't too", line++, stage >= 20);
            line(player, "fond of strangers. They will let you through if you can !!bring ", line++, stage >= 20);
            line(player, "!!them a trophy.?? The trophy is the prize for the annual Hemenster", line++, stage >= 20);
            line(player, "!!fishing competition.??", line++, stage >= 20);
            if (stage == 20)
                line(player, "I should return to !!Austri?? or !!Vestri??.", line++);

            if (stage >= 100) {
                line++;
                line(player, "<col=FF0000>QUEST COMPLETE!</col>", line++);
            }
        }
    }

    @Override
    public void finish(Player player) {
        int ln = 10;
        super.finish(player);
        player.getPacketDispatch().sendItemZoomOnInterface(FISHING_TROPHY.getId(), 230, 277, 5);
        drawReward(player, "1 Quest Point", ln++);
        drawReward(player, "2437 Fishing XP.", ln++);
        drawReward(player, "Access to the White Wolf Mountain shortcut.", ln);
        removeAttribute(player, "fishing_contest:garlic");
        removeAttribute(player, "fishing_contest:won");
        removeAttribute(player, "fishing_contest:pass-shown");
        player.getSkills().addExperience(Skills.FISHING, 2437);
    }

    @Override
    public Quest newInstance(Object object) {
        return this;
    }

}

File: Emulator/src/main/content/region/kandarin/quest/fishingcompo/FishingContest.kt
package content.region.kandarin.quest.fishingcompo

import content.data.GameAttributes
import core.api.hasLevelStat
import core.api.removeAttributes
import core.api.rewardXP
import core.api.sendItemZoomOnInterface
import core.game.node.entity.player.Player
import core.game.node.entity.player.link.quest.Quest
import core.game.node.entity.skill.Skills
import core.plugin.Initializable
import org.rs.consts.Components
import org.rs.consts.Items
import org.rs.consts.Quests

/**
 * Represents the Fishing Contest quest.
 *
 * Authors: [Woah](https://gitlab.com/woahscam), szu.
 */
@Initializable
class FishingContest : Quest(Quests.FISHING_CONTEST, 62, 61, 1, 11, 0, 1, 5) {

    /**
     * Draws the quest journal.
     */
    @OptIn(ExperimentalStdlibApi::class)
    override fun drawJournal(player: Player, stage: Int) {
        super.drawJournal(player, stage)
        var line = 11

        when (stage) {
            0 -> {
                line(player, "I can start this quest by speaking to the !!Dwarves?? at the", line++, false)
                line(player, "tunnel entrances on either side of !!White Wolf mountain??.", line++, false)
                line(player, if (hasLevelStat(player, Skills.FISHING, 10)) "---I must have level 10 fishing./--" else "!!I must have level 10 fishing??.", line++)
            }

            in 1..<100 -> {
                line(player, "The Dwarves will let me use the tunnel through White Wolf", line++, true)
                line(player, "Mountain if I can win the Hemenster Fishing Competition.", line++, true)

                if (stage >= 20) {
                    line(player, "I easily won the contest by catching some Giant Carp.", line++, false)
                } else if (stage >= 10) {
                    line(player, "They gave me a !!Fishing Contest Pass?? to enter the contest.", line++, false)
                    line(player, "I need to bring them back the !!Hemenster Fishing Trophy??.", line++, false)
                }

                if (stage >= 20) {
                    line(player, "I should take the !!Trophy?? back to the !!Dwarf?? at the side of", line++, false)
                    line(player, "!!White Wolf Mountain?? and claim my !!reward??.", line++, false)
                }
            }

            else -> {
                line(player, "The Dwarves wanted me to earn their friendship by winning", line++, true)
                line(player, "the Hemenster Fishing Competition.", line++, true)
                line(player, "I scared away a vampyre with some garlic and easily won the", line++, true)
                line(player, "contest by catching some Giant Carp.", line++, true)
                line++
                line(player, "<col=FF0000>QUEST COMPLETE!</col>", line++, false)
                line++
                line(player, "As a reward for getting the Fishing Competition Trophy the", line++, false)
                line(player, "Dwarves will let me use their tunnel to travel quickly and", line++, false)
                line(player, "safely under White Wolf Mountain anytime I wish.", line++, false)
            }
        }
    }

    /**
     * Finishes the quest.
     *
     * @param player the player completing the quest.
     */
    override fun finish(player: Player) {
        super.finish(player)
        var ln = 10
        sendItemZoomOnInterface(player, Components.QUEST_COMPLETE_SCROLL_277, 5, Items.FISHING_TROPHY_26, 230)
        drawReward(player, "1 Quest Point", ln++)
        drawReward(player, "2437 Fishing XP.", ln++)
        drawReward(player, "Access to the White Wolf Mountain shortcut.", ln)
        rewardXP(player, Skills.FISHING, 2437.0)
        removeAttributes(player, GameAttributes.QUEST_FISHINGCOMPO_STASH_GARLIC, GameAttributes.QUEST_FISHINGCOMPO_WON)
    }

    override fun newInstance(`object`: Any?): Quest = this

}
